nautobot/extras/api/serializers.py

```diff
@@ -1009,6 +1020,13 @@ class StatusModelSerializerMixin(serializers.Serializer):
 class TagSerializer(CustomFieldModelSerializer):
     url = serializers.HyperlinkedIdentityField(view_name="extras-api:tag-detail")
     tagged_items = serializers.IntegerField(read_only=True)
+    content_types = ContentTypeField(
+        queryset=ContentType.objects.filter(
+            ModelSubclassesQuery("nautobot.core.models.generics.PrimaryModel").get_query()
+        ),
+        many=True,
+        required=False,
+    )
```

Comment:
It occurs to me that this is arguably a breaking behavior change - while this field is not breaking the API itself (since you've set `required=False`), it does mean that a user creating a tag using the existing REST API gets an "unusable" Tag as a result since it's not applicable to any content-types (whereas before this changeset the created Tag would be applicable to all content-types).

For backward-compatibility, we might need some logic in the Tag clean() or save() method to special-case that if the user didn't provide any content-types, that we "fix" that to instead select all content-types? Will need to think about the UI experience though.

Reply:
content_type with no value defaults to all content_type

Reply:
I want to discuss/reconsider this design - it feels to me like a short-cut for simplified implementation at the cost of adding additional complexity for users and maintainers down the road. For example, with this design, `Tag.objects.filter(content_types__in=[my_content_type])` won't give me the expected result of all Tags that apply to my given content-type, etc.